examples/src/test/java/jdocs/testkit/TestKitDocExample.java

```diff
@@ -0,0 +1,143 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package jdocs.testkit;
+import akka.Done;
+import akka.actor.ClassicActorSystemProvider;
+import akka.projection.Projection;
+import akka.projection.ProjectionId;
+import akka.projection.ProjectionSettings;
+import akka.projection.RunningProjection;
+import akka.stream.scaladsl.Source;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+
+
+//#testkit-import
+import org.junit.ClassRule;
+import akka.actor.testkit.typed.javadsl.TestKitJunitResource;
+import akka.projection.testkit.javadsl.ProjectionTestKit;
+
+//#testkit-import
+
+//#testkit-duration
+import java.time.Duration;
+
+//#testkit-duration
+
+//#testkit-assertion-import
+import akka.stream.testkit.TestSubscriber;
+import akka.actor.testkit.typed.javadsl.TestProbe;
+import static org.junit.Assert.assertEquals;
+
+//#testkit-assertion-import
+
+public class TestKitDocExample {
+
+  static class CartView {
+    final String id;
+
+    CartView(String id) {
+      this.id = id;
+    }
+  }
+  static class CartCheckoutRepository {
+    public CompletionStage<CartView> findById(String id) {
+      return CompletableFuture.completedFuture(new CartView(id));
+    }
+  }
+
+  //#testkit
+  @ClassRule
+  static final TestKitJunitResource testKit = new TestKitJunitResource();
+  ProjectionTestKit projectionTestKit = ProjectionTestKit.create(testKit.testKit());
+  //#testkit
+
+
+
+  Projection<String> projection = new Projection<String>() {
+    @Override
+    public ProjectionId projectionId() {
+      return null;
+    }
+
+    @Override
+    public Projection<String> withSettings(ProjectionSettings settings) {
+      return null;
+    }
+
+    @Override
+    public Source<Done, ?> mappedSource(ClassicActorSystemProvider systemProvider) {
+      return null;
+    }
+
+    @Override
+    public RunningProjection run(ClassicActorSystemProvider systemProvider) {
+      return null;
+    }
+  };
+
+  CartCheckoutRepository cartCheckoutRepository = new CartCheckoutRepository();
+
+  void illustrateTestKitRun() {
+
+    //#testkit-run
+    projectionTestKit.run(projection, () -> {
+
+      TestProbe<CartView> testProbe = testKit.testKit().createTestProbe("cart-view-probe");
+      cartCheckoutRepository
+              .findById("abc-def")
+              // use a probe to capture the async result
+              .thenAccept(view -> testProbe.ref().tell(view));
```

Comment:
that can be simplifed by not using a `TestProbe`. Something like
```
CartView cartView = cartCheckoutRepository..findById("abc-def").toCompletableFuture().get(1, TimeUnit.SECONDS);
assertEquals("abc-def", cartView.id);
```

Reply:
well, I was exactly trying to avoid this nasty api, but agree it aligns better with the ScalaTest `futureValue`

Reply:
ah, I remember now another issue. 

Calling `.get()` throws exception and because we are inside a Runnable we have to handle it in place (instead of letting it bubble).

Reply:
ok, then we leave it as this now, but we should change `run` to take one of Akka's function types that allows throwing Exception

Reply:
ok, I can explore it now and send in the PR with the fix for `runWithTestSink`. For after release